pywbem/_logging.py

```diff
@@ -14,371 +14,303 @@
-LOG_OPS_CALLS_NAME = 'pywbem.ops'
+* ``pywbem.http`` - logs HTTP requests and responses between the pywbem client
+  and WBEM server. This logs the http request data and response data including
+  http headers and XML payload.
 
-#: Name of logger for HTTP requests and responses between the pywbem client and
-#: WBEM servers.
-LOG_HTTP_NAME = 'pywbem.http'
 
-#: List of the logger names that :class:`~pywbem._logging.PywbemLoggers`
-#: recognizes.
-LOG_COMPONENTS = ['ops', 'http', 'all']
+Both of the above loggers define log entries when logging is at the DEBUG log
+level.  Also, when either logger is active, each new WBEMConnection is logged.
 
-#: List of log destinations that :class:`~pywbem._logging.PywbemLoggers`
-#: recognizes.
-LOG_DESTINATIONS = ['file', 'stderr', 'none']
+There are three parts to configuration of the pywbem loggers:
 
-#: List of the log detail strings that :class:`~pywbem._logging.PywbemLoggers`
-#: recognizes.
-LOG_DETAIL_LEVELS = ['all', 'min']
+* **Configure the loggers** - This involves setting the python logging
+  parameters that control the logger and its output and may be done by either
+  python logging methods or methods within WBEMConnection. It requires defining
+  the specific loggers that will control pywbem logging.
 
-#: Default log detail level string if none is supplied with a call to the
-#: :class:`pywbem._logging.PywbemLoggers` methods that configure pywbem named
-#: loggers
-DEFAULT_LOG_DETAIL_LEVEL = 'min'
+* **Set the `detail_level` for pywbem log** - Because pywbem logs can
+  generate large quantities of information, pywbem allows the user to control
+  the quantity of information in each log.  Since this is a pywbem feature it
+  can only be configured by the pywbem :class:`~pywbem.WBEMConnection`
+  methods that configure loggers. Generally the detail levels are:
 
-#: Default maximum log entry size if none is supplied when creating
-#: :class:`~pywbem.LogOperationRecorder` objects.
-DEFAULT_MAX_LOG_ENTRY_SIZE = 1000
+  * `all` - All of the information is output.
 
-#: Default log destination if none is defined when named loggers are
-#: configured. 'none' means that there is no logging.
-DEFAULT_LOG_DESTINATION = 'none'
+  * `summary` - Only summary information is output. For api responses this
+    is primarily the count and type of objects returned. For http logs
+    the HTTP header information is output.
 
+  * `paths` - For api responses that include CIMInstanceNames or CIMClassNames,
+    (paths components in the objects) string representation of the
+    path components is output. For responses that  do not include CIMInstance
+    paths the complete objects representations are output.
 
-class MetaPywbemLoggers(type):
-    """
-    **Experimental:** *New in pywbem 0.11 as experimental.*
+  * :class:`py:int` - All of the information is output up to the size defined
+    by the integer value.  This applies to all log entries.
```

Comment:
What does "This applies to all log entries" mean: Both loggers? The sum of the sizes of all entries? The size of each entry? Request and response individually? I suggest being more specific.

Reply:
Done